### ISSUE TITLE
Qustion about implement of eq. 15 in the Paper.

### ISSUE THREAD

**Issue by Binyr** [jy0205/Pyramid-Flow#151]
Hi, thanks for your great work!

I read part of your code. I guess this line
(https://github.com/jy0205/Pyramid-Flow/blob/e4b02ef31edba13e509896388b1fedd502ea767c/pyramid_dit/pyramid_dit_for_video_gen_pipeline.py#L737) is used to calculate coefficient of Up(x_{e_{i+1}}) in formula 15, but it is different from formula 15.

Could you explain it for me (: ?

Thank you,
Yanrui

**Comment by jy0205** [comment]
Yes, this is the coefficient of Up(x_{e_{i+1}}) in formula 15.

**Comment by krunolp** [comment]
> Yes, this is the coefficient of Up(x_{e_{i+1}}) in formula 15.

Yes, but I think @Binyr meant to say that the formulas (15) and (20) in the paper do not match the formula you use in the code. In the paper, the coefficient of Up(x_{e_{i+1}}) is s_i/e_{i+1}, and by plugging in the formula for e_{i+1} from Eqn. (25) the expression is still different from the one you use in the code.

Could you please clarify how did you calculate the alpha in your code? @jy0205  
Thank you
Kruno

**Comment by feifeiobama** [comment]
Sorry for the late reply, the code computes the coefficeint of $s_k$ w.r.t. $e_{k+1}$, while the paper computes the coefficient of $e_{k+1}$ w.r.t. $s_k$. Below is the derivation of code (starting from Eq. (25) in the paper):

<img width="752" alt="Image" src="https://github.com/user-attachments/assets/e91fa0da-d643-4641-99b9-856c09baafc4" />
